src/oemof_tabular_plugins/wefe/facades/crop_v02.py

```diff
@@ -0,0 +1,556 @@
+            """function witin https://doi.org/10.1016/j.eja.2019.01.009"""
+            if t_max <= t_heat:
+                x = 1
+            elif t_heat < t_max <= t_ext:
+                x = 1 - (t_max - t_heat) / (t_ext - t_heat)
+            elif t_max > t_ext:
+                x = 0
+            return x
+
+        # Check if input arguments have proper type and length
+        if not isinstance(t_air, (list, pd.Series, np.ndarray)):
+            raise TypeError("Argument t_air is not of type list or pd.Series!")
+
+        n_timesteps = len(t_air)
+
+        if n_timesteps == 0:
+            raise ValueError("Argument t_air is empty")
+
+        n_hours_in_day = 24
+
+        n_days, n_hours = np.divmod(n_timesteps, n_hours_in_day)
+
+        # in the paper they provide T_heat as T_max in the table 1a
+        t_heat = t_max
+        hi = []  # creating a list
+
+        if n_days == 0:
+            # there is less than a day, returing same value n_timesteps times
+            n_timesteps = n_hours
+            t_max = np.max(t_air)
+            hi.append(np.ones(n_timesteps) * paper_function(t_max, t_heat, t_ext))
+        else:
+            # there is one day or more
+            if n_hours != 0:
+                N = n_days + 1
+            else:
+                N = n_days
+            for i in range(0, N):
+                if i == n_days:
+                    if n_hours != 0:
+                        # last day is not a full day
+                        n_timesteps = n_hours
+                    else:
+                        # last day is a full day
+                        n_timesteps = n_hours_in_day
+                else:
+                    # it is a full day
+                    n_timesteps = n_hours_in_day
+                t_max = np.max(
+                    t_air[(i * n_hours_in_day) : (i * n_hours_in_day + n_timesteps)]
+                )
+                hi.append(np.ones(n_timesteps) * paper_function(t_max, t_heat, t_ext))
+
+        return np.hstack(hi)
+
+    @property
+    def efficiency(self):
+
+        # Conversion factor Watt Hours (WH) to Mega Joules (MJ)
+        c_wh_to_mj = 3.6e-3
+        crop_params = crop_dict[self.crop_type]
+        fTEMP = self.calc_Ftemp(self.t_air, **crop_params)
+        fWATER = self.calc_Fwater(self.et_0, self.vwc, **crop_params)
+        fHEAT = self.calc_Fheat(t_air=self.t_air, **crop_params)
+        fSOLAR = self.calc_Fsolar(time_index=self.time_index, t_air=self.t_air,
+                                  sowing_date=self.sowing_date, harvest_date=self.harvest_date, **crop_params)
+
+        rue = self.get_crop_param("rue")
+        hi = self.get_crop_param("hi")
+
+        plt.plot(self.time_index, fTEMP, label='f_temp')
+        plt.plot(self.time_index, fWATER, label='f_water')
+        plt.plot(self.time_index, fHEAT, label='f_heat')
+        plt.plot(self.time_index, fSOLAR, label='f_solar')
+        plt.grid(True)
+        plt.legend()
+
+        plt.show()
```

Comment:
Actually plotting it with respect to Thermal time I get 

![fsolar](https://github.com/user-attachments/assets/a21dce36-b9e6-42ff-8431-fa18a9875bc3)

Reply:
The first fSOLAR looks weird because I specified sowing_date as 07-15 (start of thermal time accumulation) and harvest_date as 05-31 (end of thermal time accumulation). So it is the same curve as if you would plot to thermal time, but cultivation starts mid july in the first year and ends in may of the following year. 
Because we only have time series for one year, the beginning of that year is used instead of the following year.

I allowed three options:
1. No sowing_date and no harvest_date given: thermal time accumulation from 01-01 to 12-31, plant is finished according to thermal time for maturity (t_sum) from SIMPLE

2. Sowing_date given, no harvest_date given: thermal time accumulation from <sowing_date> to <sowing_date> - 1 (for example from 04-01 to 03-31), plant is finished according to thermal time for maturity (t_sum) from SIMPLE

3. Sowing_date and harvest_date given: thermal time accumulation from <sowing_date> to <harvest_date>, plant is finished exactly at harvest date, t_sum is updated accordingly and will differ from SIMPLE.

Reply:
With regards to the banana production, it seems that you kept the sowing_date and harvest_date and used the time series I had for morocco?

I also tested banana and if I dont specify the dates, growth starts 01-01 but looking at f_solar, it is not finished by the end of the year. That means, t_sum from SIMPLE is bigger than the updated t_sum if harvest_date is provided and even too big for a whole year of thermal time accumulation. It seems, Morocco is too cold for banana cultivation.

But for tomato, the yield was also a little low.